src/egit.c

```diff
@@ -113,7 +114,7 @@ static void egit_finalize(void* _obj)
     // Note that this object must be freed before potentially freeing owners.
     git_repository *repo = NULL;
     switch (obj->type) {
-    case EGIT_COMMIT: case EGIT_TREE: case EGIT_BLOB: case EGIT_TAG: case EGIT_OBJECT:
+    case EGIT_COMMIT: case EGIT_TREE: case EGIT_BLOB: case EGIT_TAG: case EGIT_OBJECT: case EGIT_SIGNATURE:
```

Comment:
The reason all of these are together is that `git_commit`, `git_tree`, `git_blob` and `git_tag` structs are all compatible with `git_object` and share a common interface in libgit2. This is not the case with `git_signature`. You'll need a separate case where you call `git_signature_free`. Note that a `git_signature` does not contain a reference to a repository so no need to worry about that.

Reply:
Oh, I thought `git_object` was the default object, and everything was one unless it needed to not be for some reason.